internal/pkg/initialize/workload.go

```diff
@@ -0,0 +1,317 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package initialize
+
+import (
+	"encoding"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/aws/copilot-cli/internal/pkg/term/color"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+)
+
+const (
+	fmtAddWlToAppStart    = "Creating ECR repositories for %s %s."
+	fmtAddWlToAppFailed   = "Failed to create ECR repositories for %s %s.\n"
+	fmtAddWlToAppComplete = "Created ECR repositories for %s %s.\n"
+)
+
+const (
+	jobWlType = "job"
+	svcWlType = "service"
+)
+
+var fmtErrUnrecognizedWlType = "unrecognized workload type %s"
+
+// Store represents the methods needed to add workloads to the SSM parameter store.
+type Store interface {
+	GetApplication(appName string) (*config.Application, error)
+	CreateService(service *config.Workload) error
+	CreateJob(job *config.Workload) error
+	ListServices(appName string) ([]*config.Workload, error)
+	ListJobs(appName string) ([]*config.Workload, error)
+}
+
+// WorkloadAdder contains the methods needed to add jobs and services to an existing application.
+type WorkloadAdder interface {
+	AddJobToApp(app *config.Application, jobName string) error
+	AddServiceToApp(app *config.Application, serviceName string) error
+}
+
+// Workspace contains the methods needed to manipulate a Copilot workspace.
+type Workspace interface {
+	CopilotDirPath() (string, error)
+	WriteJobManifest(marshaler encoding.BinaryMarshaler, jobName string) (string, error)
+	WriteServiceManifest(marshaler encoding.BinaryMarshaler, serviceName string) (string, error)
+}
+
+// Prog contains the methods needed to render multi-stage operations.
+type Prog interface {
+	Start(label string)
+	Stop(label string)
+}
+
+// WorkloadProps contains the information needed to represent a Workload (job or service).
+type WorkloadProps struct {
+	App            string
+	Type           string
+	Name           string
+	DockerfilePath string
+	Image          string
+}
+
+// JobProps contains the information needed to represent a Job
+type JobProps struct {
+	*WorkloadProps
```

Comment:
This package is very nice thanks! This is kinda minor but we can probably just have these as non-pointers as the data is not large and copying data should make the code safer